src/en/authors-charm-config.md

```diff
@@ -2,25 +2,26 @@ Title: Creating config.yaml and configuring charms
 
 # Charm configuration
 
-The optional `config.yaml` file defines how a service running the charm can be
-configured by its user.
+The optional `config.yaml` file defines how the charm can be configured by the
+user.
 
-If `config.yaml` exists, it must contain a dictionary of `options`, in which
-each possible setting is named by the key and defined by the value. An option
-definition must contain all of the following fields:
+If `config.yaml` is used, it must contain a dictionary of `options`, in which
+every setting is named by the key and defined by the value. An option
+definition must contain the `type` field. Otherwise, an error will be generated
+by the [charm proof tool](tools-charm-tools.html#proof) indicating the option
+is not compliant with Charm Store policy.
 
-  - `type` can be `string`, `int`, `float`, or `boolean`. The default type is
-    `string`.
-  - `description` should contain an explanation of what the user might achieve
-    by altering the setting along with valid values.
-  - `default` should contain a value of the appropriate type. If set as
-    `default:` with no trailing characters, it is treated as null (which is
-    always a valid value in this context).
+The following fields are available:
 
-Any option without these three fields will generate a Warning from the the
-[charm proof tool](tools-charm-tools.html#proof)
-indicating the option is not compliant with charm store policy. This policy
-allows older versions of juju to safely unset values.
+ - `type` : can be `string`, `int`, `float`, or `boolean`. The default type is
+  `string`.
+ - `description` : contains an explanation of what the user might achieve by
+  altering the setting along with valid values.
+ - `default` : contains a default value for the option. This value must be of
+  the appropriate type and be coherent for the given charm. If such a default
+  is mandatory for the charm to run then a value must be supplied. Null
+  values are valid for the `string` type where (`default:` and `default: ''`
```

Comment:
Empty (or null values) are valid for any option type. In fact, if the default is not defined in charm's config.yaml, it is expected that the value for this particular option will be provided by the user that will be deploying this charm in application's config.yaml (if this option is necessary for a successful deployment of this charm or successful running of the application).

Reply:
When I say 'expected', I do not mean that Juju will check that the value for this option has been specified by the user that is deploying the charm. Juju may have troubles deploying this charm or running its application.
So, it may be worthwhile to add that our charm authors may wish to be explicit when documenting the charm - "these options have default - a, b, c... But these options need to have  values explicitly stated at deployment time - x, y, z..." [because no default values are stated for these options in charm's config.yaml]

Reply:
Again, more subtlety. Juju itself may not require certain values in order to *attempt* to deploy but the charm may require them.

Reply:
How does the end user provide an option value at deploy time? Can it only be done through a separate yaml file (`juju deploy --config myconfig.yaml mysql`)?